nerfstudio/model_components/renderers.py

```diff
@@ -124,65 +128,68 @@ def get_background_color(
         elif background_color == "last_sample":
             background_color = rgb[..., -1, :]
         elif background_color == "random":
-            background_color = torch.rand_like(rgb[..., -1, :]).to(rgb.device)
+            # background_color = torch.rand_like(rgb[..., -1, :]).to(rgb.device)
+            raise RuntimeError("get_background_color should never be called with background_color='random'")
         elif isinstance(background_color, str) and background_color in colors.COLORS_DICT:
             background_color = colors.COLORS_DICT[background_color].to(rgb.device)
         assert isinstance(background_color, Tensor)
         return background_color
 
-    @classmethod
     def blend_background(
-        cls,
-        rgb: Tensor,
-        opacity: Tensor,
-        background_color: Float[Tensor, "3"],
+        self,
+        image: Tensor,
+        background_color: Optional[Float[Tensor, "3"]] = None,
     ) -> Float[Tensor, "*bs 3"]:
-        """Blends the background color into the image.
+        """Blends the background color into the image if image is RGBA.
+        Otherwise no blending is performed (we assume opacity of 1).
 
         Args:
-            rgb: RGB per pixel.
+            image: RGB/RGBA per pixel.
             opacity: Alpha opacity per pixel.
             background_color: Background color.
 
         Returns:
             Blended RGB.
         """
-        if len(opacity.shape) < len(rgb.shape):
-            opacity = opacity.unsqueeze(-1)
+        if background_color is None:
+            background_color_name = self.background_color
+            if background_color_name == "last_sample" or background_color == "random":
+                background_color_name = "black"
+            background_color = self.get_background_color(
+                rgb=image.unsqueeze(-1), background_color=background_color_name
+            )
+        if image.size(-1) < 4:
+            return image
```

Comment:
Better to move this before `if background_color is None:` to avoid unnecessary computations

Reply:
Sure, sorry